src/common/procedure/src/procedure.rs

```diff
@@ -289,6 +296,10 @@ pub enum ProcedureState {
     Done { output: Option<Output> },
     /// The procedure is failed and can be retried.
     Retrying { error: Arc<Error> },
+    /// The procedure is failed and commits state before rolling back the procedure.
+    CommitRollback { error: Arc<Error> },
```

Comment:
Maybe `PrepareRollback` is more appropriate? `CommitRollback` looks like a state after `RollingBack`.